### PR TITLE
Add methods assigned to me 

### PR DESCRIPTION
**Pull request by javier-aguilar**
- [X] Most Tackles
- [X] Least Tackles

Also includes TeamGameCollection and TeamGame classes
Plus tests

### CODE REVIEW

File: test/game_team_collection_test.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  def test_it_can_calculate_tackles
+    skip
+  end
+
+  def test_it_can_calculate_season_stats
+    skip
+  end
+
+end
```
Comment (steve-meyers) [inline test/game_team_collection_test.rb:46]: We will have to address these skipped tests at some point.

**Review by steve-meyers — APPROVED** [review]
I think the changes look good. There are several things we need to clean up but they have all been commented on in other pull requests.

File: test/game_collection_test.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    assert_instance_of Array, @game_collection.games
+    assert_equal 10, @game_collection.games.length
+  end
+
+  def test_it_can_create_games_from_csv
+    game = mock('game')
+    @game_collection.stubs(:games).returns(game)
+
+    assert_equal game, @game_collection.games
+  end
```
Comment (Shock253) [inline test/game_collection_test.rb:27]: this test is still just testing the stubs method, no code is actually checked here

File: test/game_team_collection_test.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def test_it_has_attributes
+    assert_instance_of Array, @game_team_collection.game_teams
+    assert_equal 14882, @game_team_collection.game_teams.length
+  end
+
+  def test_it_can_create_game_teams
+    skip
+  end
```
Comment (Shock253) [inline test/game_team_collection_test.rb:27]: this is an empty test

File: test/game_team_collection_test.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  def test_it_can_calculate_tackles
+    skip
+  end
+
+  def test_it_can_calculate_season_stats
+    skip
+  end
```
Comment (Shock253) [inline test/game_team_collection_test.rb:44]: so are these, did these have code in them before?

**Review by Shock253 — APPROVED** [review]
We have a lot of clutter in our branches from other branches for dependency's sake, maybe we could make better use of mocks and stubs? I'll go ahead and approve this so we can address merge conflicts later, but we'll need to be more careful about that in the future

**Pull request merged**